cdap-ui/app/directives/dag-plus/my-dag.html

```diff
@@ -204,6 +204,29 @@
                 metrics-data="metricsData">
               </my-node-metrics>
             </div>
+            <div class="node-alerts-errors"
+                  ng-if="DAGPlusPlusCtrl.shouldShowAlertsPort(node) || DAGPlusPlusCtrl.shouldShowErrorsPort(node)">
+              <div class="node-alerts"
+                    ng-if="DAGPlusPlusCtrl.shouldShowAlertsPort(node)">
+                <span>Alert</span>
+                <div class="endpoint-circle endpoint-circle-bottom"
+                      ng-class="{'disabled': DAGPlusPlusCtrl.isDisabled}"
+                      ng-mouseup="DAGPlusPlusCtrl.selectEndpoint($event, node)"
+                      ng-attr-id="{{node.name + '_alert'}}">
+                  <div class="endpoint-caret" ng-if="!DAGPlusPlusCtrl.isDisabled"></div>
+                </div>
+              </div>
+              <div class="node-errors"
+                    ng-if="DAGPlusPlusCtrl.shouldShowErrorsPort(node)">
+                <div class="endpoint-circle endpoint-circle-bottom"
+                      ng-class="{'disabled': DAGPlusPlusCtrl.isDisabled}"
+                      ng-mouseup="DAGPlusPlusCtrl.selectEndpoint($event, node)"
+                      ng-attr-id="{{node.name + '_error'}}">
```

Comment:
I think `node.name` can start with a number right? If so, this will cause a problem. Might have to prefix it with some alphabet

Reply:
Also it may contain space?

Reply:
Seems like it's a problem if we try to access with css, but passing it through document.getElementById seems to work fine